### PR TITLE
9 packages from savonet/ocaml-posix at 2.0.1

### PR DESCRIPTION
**Pull request by toots**
This pull-request concerns:
-`posix-base.2.0.1`: Base module for the posix bindings
-`posix-bindings.2.0.1`: POSIX bindings
-`posix-getopt.2.0.1`: Bindings for posix getopt/getopt_long
-`posix-signal.2.0.1`: Bindings for the types defined in <signal.h>
-`posix-socket.2.0.1`: Bindings for posix sockets
-`posix-socket-unix.2.0.1`: Bindings for posix sockets
-`posix-time2.2.0.1`: Bindings for posix time functions
-`posix-types.2.0.1`: Bindings for the types defined in <sys/types.h>
-`posix-uname.2.0.1`: Bindings for posix uname

---
* Homepage: https://github.com/savonet/ocaml-posix
* Source repo: git+https://github.com/savonet/ocaml-posix.git
* Bug tracker: https://github.com/savonet/ocaml-posix/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

File: packages/posix-base/posix-base.2.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+license: "MIT"
+homepage: "https://github.com/savonet/ocaml-posix"
+bug-reports: "https://github.com/savonet/ocaml-posix/issues"
+depends: [
+  "dune" {> "2.5"}
+  "integers"
+  "ctypes" {>= "0.12.1"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-base/posix-base.2.0.1/opam:15]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-bindings/posix-bindings.2.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "ctypes"
+  "posix-base" {= version}
+  "posix-types" {= version}
+  "posix-socket" {= version}
+  "posix-socket-unix" {= version}
+  "posix-uname" {= version}
+  "posix-getopt" {= version}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-bindings/posix-bindings.2.0.1/opam:20]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-getopt/posix-getopt.2.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  "dune-configurator"
+  "ounit2" {with-test}
+  "posix-uname" {with-test & = version}
+  "ctypes"
+  "posix-base" {= version}
+  "unix-errno"
+]
+available: os != "win32" & os != "bsd"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-getopt/posix-getopt.2.0.1/opam:21]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-signal/posix-signal.2.0.1/opam [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+bug-reports: "https://github.com/savonet/ocaml-posix/issues"
+depends: [
+  "dune" {> "2.5"}
+  "posix-base" {= version}
+  "ctypes"
+  "unix-errno"
+]
+available: os != "win32"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-signal/posix-signal.2.0.1/opam:18]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-socket-unix/posix-socket-unix.2.0.1/opam [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+depends: [
+  "dune" {> "2.5"}
+  "ctypes"
+  "posix-base" {= version}
+  "posix-socket" {= version}
+  "unix-errno"
+]
+available: os != "win32"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-socket-unix/posix-socket-unix.2.0.1/opam:19]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-socket/posix-socket.2.0.1/opam [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+license: "MIT"
+homepage: "https://github.com/savonet/ocaml-posix"
+bug-reports: "https://github.com/savonet/ocaml-posix/issues"
+depends: [
+  "dune" {> "2.5"}
+  "posix-base" {= version}
+  "ctypes"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-socket/posix-socket.2.0.1/opam:16]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-time2/posix-time2.2.0.1/opam [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+depends: [
+  "dune" {> "2.5"}
+  "ctypes"
+  "posix-base" {= version}
+  "posix-types" {= version}
+  "unix-errno"
+]
+available: os != "win32"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-time2/posix-time2.2.0.1/opam:19]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-types/posix-types.2.0.1/opam [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+bug-reports: "https://github.com/savonet/ocaml-posix/issues"
+depends: [
+  "ocaml" {>= "4.03"}
+  "dune" {> "2.5"}
+  "posix-base" {= version}
+  "ctypes"
+]
+available: os != "win32"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-types/posix-types.2.0.1/opam:18]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/posix-uname/posix-uname.2.0.1/opam [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+bug-reports: "https://github.com/savonet/ocaml-posix/issues"
+depends: [
+  "dune" {> "2.5"}
+  "ctypes"
+  "posix-base" {= version}
+  "unix-errno"
+]
+available: os != "win32"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/posix-uname/posix-uname.2.0.1/opam:17]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by toots** [comment]
Thanks! I'll address that with an upgrade of the `dune` version.

**Comment by kit-ty-kate** [comment]
Compilation fails on arm32:
```
#=== ERROR while compiling unix-errno.0.6.0 ===================================#
# context              2.2.0~alpha~dev | linux/arm32 | ocaml-base-compiler.4.14.0 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/unix-errno.0.6.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p unix-errno -j 79
# exit-code            1
# env-file             ~/.opam/log/unix-errno-7-152b5c.env
# output-file          ~/.opam/log/unix-errno-7-152b5c.out
### output ###
# File "lib/.errno.objs/native/_unknown_", line 1, characters 0-0:
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I lib/.errno.objs/byte -I lib/.errno.objs/native -I /home/opam/.opam/4.14/lib/integers -I /home/opam/.opam/4.14/lib/result -intf-suffix .ml -no-alias-deps -o lib/.errno.objs/native/errno.cmx -c -impl lib/errno.ml)
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s: Assembler messages:
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2048: Error: value of 65628 too large for field of 2 bytes at 2826
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2049: Error: value of 66201 too large for field of 2 bytes at 2828
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2050: Error: value of 66776 too large for field of 2 bytes at 2830
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2051: Error: value of 67351 too large for field of 2 bytes at 2832
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2052: Error: value of 67926 too large for field of 2 bytes at 2834
# /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2053: Error: value of 68501 too large for field of 2 bytes at 2836
```
and the tests fail on several platforms such as alpine or s390x

**Comment by toots** [comment]
> Compilation fails on arm32:
> 
> ```
> #=== ERROR while compiling unix-errno.0.6.0 ===================================#
> # context              2.2.0~alpha~dev | linux/arm32 | ocaml-base-compiler.4.14.0 | file:///home/opam/opam-repository
> # path                 ~/.opam/4.14/.opam-switch/build/unix-errno.0.6.0
> # command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p unix-errno -j 79
> # exit-code            1
> # env-file             ~/.opam/log/unix-errno-7-152b5c.env
> # output-file          ~/.opam/log/unix-errno-7-152b5c.out
> ### output ###
> # File "lib/.errno.objs/native/_unknown_", line 1, characters 0-0:
> # (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I lib/.errno.objs/byte -I lib/.errno.objs/native -I /home/opam/.opam/4.14/lib/integers -I /home/opam/.opam/4.14/lib/result -intf-suffix .ml -no-alias-deps -o lib/.errno.objs/native/errno.cmx -c -impl lib/errno.ml)
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s: Assembler messages:
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2048: Error: value of 65628 too large for field of 2 bytes at 2826
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2049: Error: value of 66201 too large for field of 2 bytes at 2828
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2050: Error: value of 66776 too large for field of 2 bytes at 2830
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2051: Error: value of 67351 too large for field of 2 bytes at 2832
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2052: Error: value of 67926 too large for field of 2 bytes at 2834
> # /opam-tmp/build_355aa0_dune/camlasm3d718b.s:2053: Error: value of 68501 too large for field of 2 bytes at 2836
> ```
> 
> and the tests fail on several platforms such as alpine or s390x

build failure on `arm32` is with `unix-errno`, a dependency of the package.

`getopt` has historically various implementation accross `libc` (so do other posix specs) so tests are bound to fail every now and them. All currently failing tests on specific arch/os have been skipped.

Thanks for checking!

**Comment by kit-ty-kate** [comment]
Let's merge this and fix arm32 later if needed. I'm really sorry for the wait. Thanks

**Pull request merged**